PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -41,56 +42,60 @@
 @step(r'"([^"]*)" is called on "([^"]*)" with:')
 def api_method_is_called(step,apiCall,nodeName):
     logger.info('%s is called on %s',apiCall,nodeName)
-    config['apiCall'] = apiCall
-    config['nodeId'] = nodeName
+    world.config['apiCall'] = apiCall
+    world.config['nodeId'] = nodeName
     arg_list = step.hashes
 
     options = {}
     api_utils.prepare_options(arg_list, options)
-    responses[apiCall] = {}
 
     api = api_utils.prepare_api_call(nodeName)
     response = api_utils.fetch_call(apiCall, api, options)
 
     assert type(response) is dict, 'There may be something wrong with the response format: {}'.format(response)
     
-    responses[apiCall] = {}
-    responses[apiCall][nodeName] = response
+    world.responses[apiCall] = {}
+    world.responses[apiCall][nodeName] = response
 
 #This method is identical to the method above, but creates a new thread
 #TODO: Find a way to pass aloe.world variable between threads to eliminate need for duplicate code
 @step(r'"([^"]*)" is called in parallel on "([^"]*)" with:')
 def threaded_call(step,apiCall,node):
     logger.info("Creating thread for {}".format(apiCall))
-    config['apiCall'] = apiCall
-    config['nodeId'] = node
+    world.config['apiCall'] = apiCall
+    world.config['nodeId'] = node
     arg_list = step.hashes
 
     options = {}
     api_utils.prepare_options(arg_list, options)
     api = api_utils.prepare_api_call(node)
 
-    def make_call(api,options):
+    def make_call(api,options,q):
         response = api_utils.fetch_call(apiCall, api, options)
+        responses = q.get()
         responses[apiCall] = {}
         responses[apiCall][node] = response
         return response
 
-    new_thread = threading.Thread(target=make_call, args=(api,options))
+    q = queue.Queue()
+    q.put(world.responses)
+    new_thread = threading.Thread(target=make_call, args=(api,options,q))
     new_thread.setDaemon(True)
     new_thread.start()
 
-    if 'threads' not in config:
-        config['threads'] = {}
-    config['threads'][apiCall] = new_thread
+    if 'threads' not in world.config:
+        world.config['threads'] = {}
+
+    world.config['threads'][apiCall] = new_thread
     #Wait 3 seconds to give node time to respond
     sleep(3)
```

Comment:
So here you wait for the node to respond and you assume that it finishes in 3 seconds?
What if didn't?
I also understood that you have threaded calls because you don't want api requests to wait for one another but they wait 3 seconds...

Look at the approved solution in this [SO post](https://stackoverflow.com/questions/6893968/how-to-get-the-return-value-from-a-thread-in-python). `async_result.get()` is blocking by the way.

I see the `sleep` is not new (we talked about this I think). Solution is in another PR?

Reply:
It's not waiting for the node to respond, I should perhaps clarify this a little better. It's there to make sure that the api calls aren't happening too quickly (specifically for the `interruptAttachToTangle` call). If there is any less than a 2 second delay, the node doesn't have enough time to respond to the one command before it receives the next one, and tests like that one fail.

Reply:
From the code I understand that there is a little race condition where `interruptAttachToTangle` is processed before `attachToTangle`.

I think putting the `sleep` here is very implicit. You may want to reuse this step without the sleep IIUC.
Why don't you add a step `def wait_for(step, seconds)`?